docs/articles/documentation/using-testcafe/common-concepts/browsers/testing-in-headless-mode.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: docs
+title: Testing in Headless Mode
+permalink: /documentation/using-testcafe/common-concepts/browsers/testing-in-headless-mode.html
+---
+# Testing in Headless Mode
+
+TestCafe allows you to run tests in Google Chrome and Mozilla Firefox without any visible UI shell - in the headless mode ([Chrome Headless](https://developers.google.com/web/updates/2017/04/headless-chrome), [Firefox Headless](https://developer.mozilla.org/en-US/Firefox/Headless_mode)). Use the `headless` browser parameter to launch a browser in the headless mode.
+
+```sh
+testcafe "chrome:headless" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('chrome:headless')
+    .run();
+```
+
+Specify a path to the browser executable if you use a portable version of the browser:
+
+```sh
+testcafe "firefox:path/to/firefox:headless" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('firefox:path/to/firefox:headless')
+    .run();
+```
+
+Note that the [Take screenshot](../../../test-api/actions/take-screenshot.md) and [Resize window](../../../test-api/actions/resize-window.md) features are fully supported in the headless mode.
+
+> In the headless mode, the browser's viewport size is `1280x800` by default. You can change it via the [ResizeWindow](../../../test-api/actions/resize-window.md) command or via [Device Emulation Mode](using-chrome-device-emulation.md#emulator-parameters) parameters (for Chrome).
```

Comment:
It's default resolution in Headless Chrome. We have to setup it explicitly in TestCafe because browser will not work otherwise. Headless Firefox doesn't require explicit resolution setup, so this setting is controlled by browser. On my machine it's like `1366x768`. Maybe we should force Chrome default value For Firefox? \cc @AlexanderMoskovkin

Reply:
I suppose we shouldn't. And I'm not sure that we should leave this note. A user actually doesn't know what the size a regular (non headless) browser has when it's opened. There are no differences with headless browsers. If you need to have a specified size you set it via the ResizeWindow action
/cc @VasilyStrelyaev

Reply:
okay, I see